cmd/controller-manager/app/controller-manager.go

```diff
@@ -93,16 +93,18 @@ member clusters and do the necessary reconciliation`,
 		},
 	}
 
-	// Add the command line flags from other dependencies(klog, kubebuilder, etc.)
-	cmd.Flags().AddGoFlagSet(flag.CommandLine)
-
-	opts.AddFlags(cmd.Flags())
-	cmd.Flags().StringVar(&healthzAddr, "healthz-addr", healthzDefaultBindAddress, "The address the healthz endpoint binds to.")
-	cmd.Flags().StringVar(&metricsAddr, "metrics-addr", metricsDefaultBindAddress, "The address the metric endpoint binds to.")
-	cmd.Flags().BoolVar(&verFlag, "version", false, "Prints the Version info of controller-manager.")
-	cmd.Flags().StringVar(&kubeFedConfig, "kubefed-config", "", "Path to a KubeFedConfig yaml file. Test only.")
-	cmd.Flags().StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
-	cmd.Flags().StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	flags := cmd.Flags()
+	opts.AddFlags(flags)
+	flags.StringVar(&healthzAddr, "healthz-addr", healthzDefaultBindAddress, "The address the healthz endpoint binds to.")
+	flags.StringVar(&metricsAddr, "metrics-addr", metricsDefaultBindAddress, "The address the metric endpoint binds to.")
+	flags.BoolVar(&verFlag, "version", false, "Prints the Version info of controller-manager.")
+	flags.StringVar(&kubeFedConfig, "kubefed-config", "", "Path to a KubeFedConfig yaml file. Test only.")
+	flags.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	flags.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+
+	local := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	klog.InitFlags(local)
+	flags.AddGoFlagSet(local)
```

Comment:
The pattern I've seen adopted more in https://github.com/kubernetes/kubernetes repo is like:

```
# First line of main
klog.InitFlags(nil)
...
# Do other flag stuff
...
cmd.AddGoFlagSet(flag.CommandLine)
```

Reply:
The pattern shared in this PR comes from the Kubernetes repo. Check how the initialize klog flags.

Reply:
Could you detail what you see wrong in the current pattern?

Reply:
Stupid question: Is there a difference between first initializing klog flags and then others as opposed to doing it the other way around? If not, I'm fine either way.

Reply:
Yeah either work, I was just pointing the more common approach (see https://github.com/kubernetes/kubernetes/search?q=%22klog.InitFlags%22) but it's not a biggie.